File: apps/mobile_ios/test/parkrun_regions_test.dart
import 'package:flutter_test/flutter_test.dart';

import '../lib/parkrun_regions.dart';

void main() {
  group('parkrunLikelyUnavailable', () {
    test('parkrun regions show no hint', () {
      expect(parkrunLikelyUnavailable('en-US'), isFalse);
      expect(parkrunLikelyUnavailable('en-GB'), isFalse);
      expect(parkrunLikelyUnavailable('de-DE'), isFalse);
      expect(parkrunLikelyUnavailable('ja-JP'), isFalse);
      expect(parkrunLikelyUnavailable('en-AU'), isFalse);
      expect(parkrunLikelyUnavailable('af-ZA'), isFalse);
    });

    test('regions outside the parkrun footprint show the hint', () {
      expect(parkrunLikelyUnavailable('es-ES'), isTrue);
      expect(parkrunLikelyUnavailable('fr-FR'), isTrue);
      expect(parkrunLikelyUnavailable('pt-BR'), isTrue);
      expect(parkrunLikelyUnavailable('id-ID'), isTrue);
      expect(parkrunLikelyUnavailable('zh-Hant-TW'), isTrue);
    });

    test('accepts POSIX underscore tags', () {
      expect(parkrunLikelyUnavailable('en_GB'), isFalse);
      expect(parkrunLikelyUnavailable('es_ES'), isTrue);
    });

    test('an unknown or region-less locale shows no hint (no false warning)', () {
      expect(parkrunLikelyUnavailable('en'), isFalse);
      expect(parkrunLikelyUnavailable(''), isFalse);
      expect(parkrunLikelyUnavailable('not-a-locale!!'), isFalse);
    });

    test('region codes are two-letter uppercase', () {
      for (final r in parkrunRegions) {
        expect(RegExp(r'^[A-Z]{2}$').hasMatch(r), isTrue, reason: r);
      }
    });
  });
}

File: apps/web/src/lib/integrations/parkrun_regions.test.ts
import { test } from 'node:test';
import { strict as assert } from 'node:assert';
import { parkrunLikelyUnavailable, PARKRUN_REGIONS } from './parkrun_regions';

test('parkrun regions show no hint', () => {
	assert.equal(parkrunLikelyUnavailable('en-US'), false);
	assert.equal(parkrunLikelyUnavailable('en-GB'), false);
	assert.equal(parkrunLikelyUnavailable('de-DE'), false);
	assert.equal(parkrunLikelyUnavailable('ja-JP'), false);
	assert.equal(parkrunLikelyUnavailable('en-AU'), false);
	assert.equal(parkrunLikelyUnavailable('af-ZA'), false);
});

test('regions outside the parkrun footprint show the hint', () => {
	assert.equal(parkrunLikelyUnavailable('es-ES'), true);
	assert.equal(parkrunLikelyUnavailable('fr-FR'), true);
	assert.equal(parkrunLikelyUnavailable('pt-BR'), true);
	assert.equal(parkrunLikelyUnavailable('id-ID'), true);
	assert.equal(parkrunLikelyUnavailable('zh-Hant-TW'), true);
});

test('an unknown or region-less locale shows no hint (no false warning)', () => {
	assert.equal(parkrunLikelyUnavailable('en'), false);
	assert.equal(parkrunLikelyUnavailable(''), false);
	assert.equal(parkrunLikelyUnavailable('not-a-locale!!'), false);
});

test('region codes are two-letter uppercase', () => {
	for (const r of PARKRUN_REGIONS) {
		assert.match(r, /^[A-Z]{2}$/);
	}
});
